[mod] simple theme: autocompletion doesn't use the style attribute.

dalf opened the pull request:
## What does this PR do?

Simple theme: change the autocompletion css and js not to use the style attribute.

The implementation has different issues:
* the right border can have a small gap.
* when the mouse is over the magnifier on the right, the 

![image](https://user-images.githubusercontent.com/1594191/138333191-7a70f9ae-542e-4c0c-8522-a3b1007ea32c.png)

![image](https://user-images.githubusercontent.com/1594191/138333336-1cfc3e84-bf15-4895-87b7-fd68e74e6166.png)

![image](https://user-images.githubusercontent.com/1594191/138333989-4f7a0945-c180-4235-a335-af9b0e5f2c5d.png)

## Why is this change important?

* remove `style` attributes.
* if the two above issues are fixed, cleaner UI.

## How to test this PR locally?

* `make themes`
* `make run`

## Author's checklist

<!-- additional notes for reviewiers -->

## Related issues

close #352

dalf committed:
[mod] simple theme: autocompletion doesn't use the style attribute.

close #352

dalf commented on searx/static/themes/simple/src/less/autocomplete.less:
```diff
@@ -2,10 +2,13 @@
 
 .autocomplete {
   position: absolute;
+  width: 44.7rem;
```
It should be 44,6rem = 40rem (#q) + 1.8rem (#clear_search) + 2.8rem (#send_search)
In Firefox, it looks better with 44.7rem, but not with Chrome ==> it should be implemented in a different way.

return42 replied:
Anther issue: the 40rem ...

https://github.com/searxng/searxng/blob/33e2599b8217e263c744fc14f2c89118ef363866/searx/static/themes/simple/src/less/search.less#L133

on phone ..

https://github.com/searxng/searxng/blob/33e2599b8217e263c744fc14f2c89118ef363866/searx/static/themes/simple/src/less/search.less#L219

is changed to 100% ..

https://github.com/searxng/searxng/blob/33e2599b8217e263c744fc14f2c89118ef363866/searx/static/themes/simple/src/less/search.less#L242-L243

mrpaulblack replied:
I think on phones the google approach is best... Let the autocomple take up the entire screen and move the search input form to the top... This way it does not matter where the search form is normally on the page and we can show the maximum amount of autocomplete suggestions to the user.
![image](https://user-images.githubusercontent.com/38733246/144741532-f236e46b-b054-4d06-987b-576a66e29edc.png)

dalf commented on searx/static/themes/simple/src/less/autocomplete.less:
```diff
@@ -50,9 +53,15 @@
+    z-index: 100000000;
+
+    li:first-child {
+      margin-top: 0.475rem;
+      border-top: 1px solid var(--color-autocomplete-border);
```
trick to hide the bottom round corners of #q , but there is a glitch with #send_search:hover

dalf commented on searx/static/themes/simple/src/js/main/searx_search.js:
```diff
@@ -67,13 +67,10 @@
           MinChars: 4,
           Delay: 300,
+          _Position:function() {
+            this.DOMResults.setAttribute("class", "autocomplete");
```
override https://github.com/autocompletejs/autocomplete.js/blob/87069524f3b95e68f1b54d8976868e0eac1b2c83/src/autocomplete.ts#L381-L384 

```typescript
        _Position:function(): void {
            this.DOMResults.setAttribute("class", "autocomplete");
            this.DOMResults.setAttribute("style", "top:" + (this.Input.offsetTop + this.Input.offsetHeight) + "px;left:" + this.Input.offsetLeft + "px;width:" + this.Input.clientWidth + "px;");
        },
```

mrpaulblack commented on searx/static/themes/simple/src/less/autocomplete.less:
```diff
@@ -2,10 +2,13 @@
   text-align: left;
-  .rounded-corners;
+  border-bottom-left-radius: 10px;
+  border-bottom-right-radius: 10px;
```
I think it would be cleaner to add this as a mixin: ```.rounded-bottom-corners;``` and do this in `mixins.less`:
```less
.rounded-bottom-corners (@radius: 10px) {
  -webkit-border-bottom-left-radius: @radius;
  -webkit-border-bottom-right-radius: @radius;
  -moz-border-bottom-left-radius: @radius;
  -moz-border-bottom-right-radius: @radius;
  border-bottom-left-radius: @radius;
  border-bottom-right-radius: @radius;
}
```
(I am not sure if the border-bottom property actually exists for webkit and moz tho)

return42 replied:
FYI `-webkit-border...-radius` (28a597e43e) and ` -moz-border...-radius` (8935b1d) are no longer needed.

> I think it would be cleaner to add this as a mixin: ```.rounded-bottom-corners;``` and do this in `mixins.less`

Thanks for the hint, I will implement this today .. I still working on this PR .. coming soon ... see 2981102cb8

return42 committed:
[mod] move border radius of .autocomplete to a LESS mixin

Create `.rounded-bottom-corners()` mixin in less/mixins.less

Suggested-by: @mrpaulblack  https://github.com/searxng/searxng/pull/426#pullrequestreview-786137688
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
